index.bs

```diff
@@ -1153,10 +1153,10 @@ run the following steps:
 1. If |domain| is not null, then run these steps:
     1. If |domain| starts with U+002E (.), then return failure.
     1. If |name|, [=byte-lowercased=], [=byte sequence/starts with=] \``__host-`\`, then return failure.
-    1. If |host| does not equal |domain| and
-        |host| does not end with U+002E (.) followed by |domain|,
-        then return failure.
-    1. Let |encodedDomain| be the result of [=UTF-8 encode|UTF-8 encoding=] |domain|.
+    1. If |domain| [=is a registrable domain suffix of or is equal to|is not a registrable domain suffix of and is not equal to=] |host|, then return failure.
+    1. Let |parsedDomain| be the result of [=host parser|host parsing=] |domain|.
+    1. If |parsedDomain| is a failure, then return failure.
```

Comment:
Can we actually hit this step? Wouldn't the first algorithm have returned failure for this case? I think this should be an assert instead.